autorest/azure/async_test.go

```diff
@@ -812,6 +812,50 @@ func TestDoPollForAsynchronous_ReturnsAnErrorForFailedOperations(t *testing.T) {
 		autorest.ByClosing())
 }
 
+func TestDoPollForAsynchronous_ReturnsAnUnknownErrorForFailedOperations(t *testing.T) {
+	r1 := newAsynchronousResponse()
+	r1.Header.Del(http.CanonicalHeaderKey(headerAsyncOperation))
+	r2 := newProvisioningStatusResponse("busy")
+	r2.Header.Del(http.CanonicalHeaderKey(headerAsyncOperation))
+	r3 := newProvisioningStatusResponse(operationFailed)
+	r3.Header.Del(http.CanonicalHeaderKey(headerAsyncOperation))
+
+	client := mocks.NewSender()
+	client.AppendResponse(r1)
+	client.AppendAndRepeatResponse(r2, 2)
+	client.AppendAndRepeatResponse(r3, 1)
+
+	r, err := autorest.SendWithSender(client, mocks.NewRequest(),
+		DoPollForAsynchronous(time.Millisecond))
+
+	if err == nil || (!strings.Contains(fmt.Sprintf("%v", err), "Unknown") && !strings.Contains(fmt.Sprintf("%v", err), "None")) {
```

Comment:
why not:

```
if err == nil {
 t.Fatal("was expecting error; got nil")
else if err.Error() != expected {
  t.Fatalf("wrong error message: got=%q expected=%q...)
}
```

we should switch to gocheck for this repo as well, that'd dramatically shorten the test code.